### PR TITLE
Added support SAP_VENDOR_ENABLED for DateTimeOffset

### PR DESCRIPTION
**Pull request by AlapanGhosh**
Added support to switch between generic and SAP OData implementation for the EDMtype DateTimeOffset.  

This is an extension to #106 

Further reading: [https://blogs.sap.com/2017/01/05/date-and-time-in-sap-gateway-foundation/](https://blogs.sap.com/2017/01/05/date-and-time-in-sap-gateway-foundation/)

### CODE REVIEW

**Comment by phanak-sap** [comment]
btw, just for an info, latest version pf pyodata wil have fix for DateTimeOffset - relevant to timezones. Our odata services (sap gateway) returns everything in UTC, so far it was not a problem.

But worth looking into the timezone part in generator.py - as separate issue, since depends on version bump.

https://github.com/SAP/python-pyodata/pull/184

**Comment by AlapanGhosh** [comment]
> btw, just for an info, latest version pf pyodata wil have fix for DateTimeOffset - relevant to timezones. Our odata services (sap gateway) returns everything in UTC, so far it was not a problem.
> 
> But worth looking into the timezone part in generator.py - as separate issue, since depends on version bump.
> 
> [SAP/python-pyodata#184](https://github.com/SAP/python-pyodata/pull/184)

@phanak-sap from what I understood, the timezone details would be a separate property of the DateTimeOffset object with the new update. Since we are generating the timezone offset in generator itself, I dont think this adds a constraint on ODFuzz.

File: odfuzz/generators.py [hunk 1]
```diff
@@ -11,7 +11,11 @@
 from odfuzz.config import Config
 
 START_DATE = datetime.datetime(1970, 1, 1, 0, 0, 0)
-END_DATE = datetime.datetime(9999, 12, 31, 23, 59, 59)
+END_DATE = datetime.datetime(3000, 12, 31, 23, 59, 59)
```
Comment (phanak-sap) [inline odfuzz/generators.py:14]: 9999 is valid SAP input sadly.. 

But we can go in a way that the START-END range will be 1970 - 3000 + the timestamp value for 9999-12-31-23-59-59 (= "Infinity" in SAP world, for example validity period end) will be hardcoded String.. it is in the end used in GET/POST string anyway.

Then the generator do not need to tackle real DateTime object with such high value and go OK on LInux and fail on Windows.

Comment (AlapanGhosh) [reply odfuzz/generators.py:14]: Implemented.

**Comment by phanak-sap** [comment]
Looks good, thx for the fix.

**Pull request merged**